platform/darwin/src/MGLMapSnapshotter.h

```diff
@@ -0,0 +1,105 @@
+#import <Foundation/Foundation.h>
+#import "MGLTypes.h"
+#import "MGLGeometry.h"
+#import "MGLMapCamera.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+MGL_EXPORT
+/**
+ The options to use when creating images with the `MGLMapsnapshotter`.
+ */
+@interface MGLMapSnapshotOptions : NSObject
+
+/**
+ Creates a set of options with the minimum required information
+ @param styleURL the style url to use
+ @param camera the camera settings
+ @param size the image size
+ */
+- (instancetype)initWithStyleURL:(NSURL*)styleURL camera:(MGLMapCamera*)camera size:(CGSize)size;
+
+#pragma mark - Configuring the map
+
+/**
+  The style URL for these options.
+ */
+@property (nonatomic, readonly) NSURL* styleURL;
+
+/**
+ The zoom. Default is 0.
+ */
+@property (nonatomic) double zoom;
+
+/**
+ The `MGLMapcamera` options to use.
+ */
+@property (nonatomic) MGLMapCamera* camera;
+
+/**
+ A region to capture. Overrides the center coordinate
+ in the mapCamera options if set
+ */
+@property (nonatomic) MGLCoordinateBounds region;
+
+#pragma mark - Configuring the image
+
+/**
+ The size of the output image. Minimum is 64x64
+ */
+@property (nonatomic, readonly) CGSize size;
+
+/**
+ The scale of the output image. Defaults to the main screen scale.
+ Minimum is 1.
+ */
+@property (nonatomic) CGFloat scale;
+
+@end
+
+/**
+ A block to processes the result or error of a snapshot request.
+ 
+ The result will be either an `MGLImage` or a `NSError`
```

Comment:
`MGLImage` is an unfamiliar typedef that we don’t document anywhere. Instead, create duplicate declarations, one using NSImage and the other using UIImage, and have the compiler switch between them using conditional compilation. See MGLStyle.h for an example.

Reply:
This code is similar to `MGLStyle.h` https://github.com/mapbox/mapbox-gl-native/blob/master/platform/darwin/src/MGLStyle.h#L523 is this ok?

Reply:
There’s nothing technically wrong with using `MGLImage` as the return value, since the compiler (and thus jazzy) will automatically substitute NSImage or UIImage as appropriate. But neither Quick Help nor jazzy will replace `MGLImage` inside a documentation comment. The same goes for `MGLColor`; that’s why we [use conditional compilation](https://github.com/mapbox/mapbox-gl-native/blob/53d072cbf539b30301fe3b77336751bf2c29af89/platform/darwin/src/MGLLineStyleLayer.h#L243-L299) whenever we need to refer specifically to the underlying type in documentation.

Besides, this sentence isn’t accurate: the block doesn’t return anything (`void`), so there’s no result to speak of. Rather, I think the intention was to say, “Either `snapshot` or `error` will be non-nil,” but that’s already suitably captured in the following two `@param` lines. So I think this line can simply be removed.

Reply:
What I meant is since `MGLStyle.h` is not using conditional compilation I should go ahead and add it there too.